sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/grouping.scala

```diff
@@ -231,3 +229,24 @@ object GroupingID {
     if (SQLConf.get.integerGroupingIdEnabled) IntegerType else LongType
   }
 }
+
+object GroupingAnalytics {
+  def unapply(exprs: Seq[Expression])
+  : Option[(Seq[Seq[Expression]], Seq[Seq[Expression]], Seq[Expression])] = {
+    val (groupingSetExprs, others) = exprs.partition(_.isInstanceOf[BaseGroupingSets])
```

Comment:
does the order matter? e.g. `GROUP BY a, GROUPING SETS((b))` and `GROUP BY GROUPING SETS((b)), a`, are they the same?

Reply:
> does the order matter? e.g. `GROUP BY a, GROUPING SETS((b))` and `GROUP BY GROUPING SETS((b)), a`, are they the same?

The result should be same, the only concern is the result order. I have checked before that 
`group by a, b` is same result as `group by b, a`. So here also can't be a concern

Reply:
And how about current code here, It can keep order when expand grouping sets.